internal/scanner/scanner.go

```diff
@@ -52,13 +54,57 @@ type Scanner struct {
 	printJSON  bool
 }
 
-// NewScanner creates a new scanner with the PoliciesFetcher provided
-func NewScanner(policiesFetcher PoliciesFetcher, resourcesFetcher ResourcesFetcher, printJSON bool) (*Scanner, error) {
+// NewScanner creates a new scanner with the PoliciesFetcher provided. If
+// insecureClient is false, it will read the caCertFile and add it to the in-app
+// cert trust store. This gets used by the httpCLient when connection to
+// PolicyServers endpoints.
+func NewScanner(policiesFetcher PoliciesFetcher, resourcesFetcher ResourcesFetcher,
+	printJSON bool,
+	insecureClient bool, caCertFile string) (*Scanner, error) {
 	report, err := report.NewPolicyReportStore()
 	if err != nil {
 		return nil, err
 	}
-	return &Scanner{policiesFetcher, resourcesFetcher, *report, http.Client{}, printJSON}, nil
+
+	// configure client for TLS
+	var httpClient http.Client
+	if insecureClient {
+		tr := &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}} //nolint
+		httpClient = http.Client{Transport: tr}
+		log.Info().Msg("connecting to PolicyServers endpoints in insecure SSL mode")
+	} else {
+		// Get the SystemCertPool, continue with an empty pool on error
+		rootCAs, _ := x509.SystemCertPool()
+		if rootCAs == nil {
+			rootCAs = x509.NewCertPool()
+		}
+		// Read in the cert file
+		certs, err := os.ReadFile(caCertFile)
+		if err != nil {
+			log.Fatal().Msg(fmt.Sprintf("failed to read file %q with CA cert: %v", caCertFile, err))
+		}
+		// Append our cert to the inherited system pool
+		if ok := rootCAs.AppendCertsFromPEM(certs); !ok {
+			log.Fatal().Msg("failed to append cert to in-app RootCAs trust store")
+		}
+		log.Debug().Str("ca-cert-file", caCertFile).
+			Msg("appended cert file to in-app RootCAs trust store")
+		// Trust the augmented cert pool in our client
+		config := &tls.Config{
+			InsecureSkipVerify: false,
+			RootCAs:            rootCAs,
+			MinVersion:         tls.VersionTLS12,
```

Comment:
I would drop that to ensure we use the minimum version recommended by the Go developers. Right now the minimum tls version is 1.2 on the client side, but this could be raised in the future because of security issue. My hope is that, by sticking to the default, we will not have to worry about possible changes in this area

Reply:
If left unset, there's no default. `gosec` complains with `G402: TLS MinVersion too low. (gosec)`. Hence, I decided to set the same MinVersion we set in the controller when creating the ca cert.
I hope that by keeping golangci-lint up to date we will get those future checks.

Reply:
I see, let's be explicit about the TLS version to be used.

As for golangci-lint, the current situation sucks (I'm referring to the Makefile), but at least the version used by the CI should be kept updated by dependabot/renovate bot. Isn't it?